### ISSUE TITLE
Double inclusion of config.asset_host in CSS `url()` used with .erb asset path helpers

### ISSUE THREAD

**Issue by jrochkind** [rails/sprockets-rails#488]
I believe this began with release 3.3.0, and is a regression caused by #476 

Let's say we have an `application.css.erb` with this in it:

```
@font-face {
  src: url('<%= font_path('example.eot') %>');
}
```

And we have `config.asset_host = '//example.org'`. (The `//` protocol-agnostic asset_host is a common setting for CDNs, often recommended in docs). 

With sprockets-rails 3.3.0+, if you `./bin/rails assets:precompile`, the output at `./public/assets/application-[fingerprint].css` looks like:

```
@font-face {
  src: url(//example.com/example.com/fonts/example.eot);
}
```

Note that `//example.com/example.com` is there twice.  This reproduces with sprockets rails 3.3.0 and 3.4.1. (both with sprockets 4.0.2). 

If you downgrade to sprockets 3.2.2, the output is as expected:

```
@font-face {
  src: url('//example.com/fonts/example.eot');
}
```

I believe this is a regression beginning with sprockets-rails 3.3.0. 

I actually ran into this using [font-awesome-rails](https://github.com/bokmann/font-awesome-rails) on an app that deploys to heroku, with a CDN `config.asset_host`.  font-awesome-rails uses the `font_path` helper in a `.css.erb` [here](https://github.com/bokmann/font-awesome-rails/blob/master/app/assets/stylesheets/font-awesome.css.erb#L14). While that font-awesome-rails gem may be no longer supported, we ought not to get a regression in a minor sprockets-rails release. But also, I then isolated the reproduction as above, which also seems to be a general problem. It may be possible to isolate/generalize further, may also occur with sass-rails helpers not just the erb ones?  May occur with other helpers, not just `font_path`?

**Comment by jrochkind** [comment]
Verified problem still reproduces (problem is still there) on sprockets-rails main branch at 4e0f1686ce10

**Comment by jrochkind** [comment]
Verified that changing `config.asset_host` to specify a protocol (eg `https://example.com`) instead of being protocol-agnostic (eg `//example.com`) avoids the problem, does not reproduce. 

So that's a workaround. 

Additionally, specifying `config.asset_host` as _just_ the bare hostname without leading `//` also seems to work. `config.host = "example.com"`, and the produced CSS output actually still has the protocol-relative `//` added to it: `src: url(//example.com/fonts/example.eot);`

So that's another workaround. 

However, I still believe this is a regression. Existing apps may use a `config.asset_host` value that begins with `//`. The value beginning `//` can be found in various examples/guides online (eg [here](https://gist.github.com/mm53bar/4171061)), and has worked in the past. 

But looks like easy workaround for anyone else that arrives here!

**Comment by jrochkind** [comment]
Workarounds?

Adding a `//` to the beginning of your `config.asset_host` isn't actually what's recommended in Rails docs ([eg](https://api.rubyonrails.org/v6.1/classes/ActionView/Helpers/AssetUrlHelper.html)).

Still, it worked before, and is recommended in some online examples ([eg](https://gist.github.com/mm53bar/4171061)), and I believev that breaking it in a minor release is still a regression. 

If we set the `config.asset_host` value to a straight hostname like `"example.com"`, then what happens? In my simple isolated reproduction... the problem does not exhibit. The CSS output still has protocol-relative value `"//example.com"`, and it's not doubled. 

But in my actual app... the problem still exhibits. I haven't figured out what the difference is or how to isolate it.  Have spent the afternoon on it and haven't given up isolating it yet. But for now I have just locked my app to sprocketes-rails 3.2.2. 

What if you set `config.asset_host` to a full url, like `"https://example.com"`?  I find problem does not exhbit (in my isolated repro app or my real app).  But that's maybe kinda bad form, and still not what Rails docs recommend. 

Anyway, if you find yourself here looking for a workaround, one of those might work.

**Comment by jrochkind** [comment]
(Sorry, I keep making comments and then deleting them about whether different forms of absolute/relative url in `config.assets_host` exhibit or not, because I keep getting confused about what i'm observing. I am also not currently sure if it exhibits on main branch, sorry!).

**Comment by jrochkind** [comment]
OK, now I think maybe the problem **IS** fixed at main branch at 4e0f168. 

Could someone else verify this though? I don't trust myself, I keep seeing different things!

If it is fixed on main, could we get a patch release soon, to spare others this trouble?

**Comment by dhh** [comment]
Fixed in https://github.com/rails/sprockets-rails/releases/tag/v3.4.2.

**Comment by mattbrictson** [comment]
I ran into this as well, and can confirm it is fixed in 3.4.2.

FWIW, for some reason I had to bump my `config.assets.version` in order to see the fix.

**Comment by jrochkind** [comment]
Confirmed fixed for me too in 3.4.2, with `config.asset_host = "//example.com"`, and also `config.asset_host = "example.com"`. 

I _think_ the second one (hostname-only) triggered bug previously as well, and is recommended in some Rails docs. 

Both seem fixed to me in 3.4.2.
